composer/algorithms/low_precision_groupnorm/low_precision_groupnorm.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Low Precision GroupNorm."""
+
+from __future__ import annotations
+
+import logging
+import warnings
+from typing import Dict, Optional, Sequence, Type, Union
+
+import torch
+import torch.nn.functional as F
+
+from composer.algorithms.warnings import NoEffectWarning
+from composer.core import Algorithm, Event, Precision, State
+from composer.loggers import Logger
+from composer.utils import module_surgery
+
+log = logging.getLogger(__name__)
+
+
+def _cast_if_autocast_enabled(hidden_states):
+    if not torch.is_autocast_enabled():
+        return hidden_states
+    else:
+        return torch.cuda.amp.autocast_mode._cast(hidden_states, torch.get_autocast_gpu_dtype())
+
+
+class LPGroupNorm(torch.nn.GroupNorm):
+
+    def __init__(self, layer):
+        super().__init__(
+            num_groups=layer.num_groups,
+            num_channels=layer.num_channels,
+            eps=layer.eps,
+            affine=layer.affine,
+        )
+
+        with torch.no_grad():
+            self.weight.copy_(layer.weight)
+            self.bias.copy_(layer.bias)
+
+    def forward(self, x):
+        module_device = x.device
+        downcast_x = _cast_if_autocast_enabled(x)
+        downcast_weight = _cast_if_autocast_enabled(self.weight)
+        downcast_bias = _cast_if_autocast_enabled(self.bias)
+        with torch.autocast(enabled=False, device_type=module_device.type):
+            return F.group_norm(downcast_x, self.num_groups, downcast_weight, downcast_bias, self.eps)
+
+
+def to_LPGroupNorm(layer: torch.nn.Module, module_index: int) -> LPGroupNorm:
+    assert isinstance(layer,
+                      torch.nn.GroupNorm), 'The replacement policy will look for all instances of torch.nn.GroupNorm'
```

Comment:
```suggestion
    message = 'The replacement policy will look for all instances of torch.nn.GroupNorm'
    assert isinstance(layer, torch.nn.GroupNorm), message
```
as long as it's two lines either way, might as well be prettier